chore: update ProviderWidget tooltips

SoniaSandler opened the pull request:
Signed-off-by: Sonia Sandler <ssandler@redhat.com>

### What does this PR do?
This PR updated the tooltips of the ProviderWidget to be one tooltip with the icon, status, and provider names

### Screenshot / video of UI
[Screencast from 2025-02-18 13-33-54.webm](https://github.com/user-attachments/assets/62b32e9c-98d4-4882-a4cd-dc1ddb6cb4c6)


<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

### What issues does this PR fix or reference?
Fixes https://github.com/podman-desktop/podman-desktop/issues/10921

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

- [X] Tests are covering the bug fix or the new feature

axel7083 commented on packages/renderer/src/lib/statusbar/ProviderWidgetStatus.ts:
```diff
@@ -0,0 +1,48 @@
+      return 'Stopping';
+    case 'stopped':
+      return 'Off';
+    case 'unknown':
+      return 'Unknown';
```
Should be ignore in favour of the default; case

axel7083 commented on packages/renderer/src/lib/statusbar/ProviderWidgetStatus.ts:
```diff
@@ -0,0 +1,48 @@
+
+export function getStatusName(status: ProviderStatus | ProviderConnectionStatus): string {
+  switch (status) {
+    case 'ready':
+      return 'Running';
+    case 'started':
+      return 'Running';
```
```suggestion
    case 'ready':
    case 'started':
      return 'Running';
```

axel7083 commented on packages/renderer/src/lib/statusbar/ProviderWidget.svelte:
```diff
@@ -14,31 +16,63 @@ interface Props {
-  let tooltip = '';
+let connectionsStatuses = $derived.by(() => {
+  let statuses: ProviderStatus | ProviderConnectionStatus[];
+  let connectionsStatuses: { status: ProviderStatus | ProviderConnectionStatus; connecions: string }[] = [];
   if (entry.containerConnections.length > 0) {
-    tooltip = entry.containerConnections.map(c => c.name).join(', ');
+    statuses = [...new Set(entry.containerConnections.map(connection => connection.status))];
+    let connections = '';
+    for (const status of statuses) {
```
Set is iterable [Set#Symbol.iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/Symbol.iterator)

So you can do the following

````ts
const statuses: Set<string> = new Set(entry.containerConnections.map(connection => connection.status)))
for(const status of statuses) {
	....
}
````

axel7083 replied:
![image](https://github.com/user-attachments/assets/5d7a64ae-fbf2-4315-bbda-a90417bc4429)

I was curious, and I used `https://jsbench.me/` to compare if spreading the set was faster/slower than using the build in iterator, result: we should use the internal iterator!

axel7083 reviewed (commented):
This PR will conflict with https://github.com/podman-desktop/podman-desktop/pull/11036

What I would suggest:

- A props should exist to disable the tooltip

deboer-tim commented:
I find the grouping by status and changing/different format (single vs multiline) a bit odd. If I have two or more connections, the most common use for the tooltip is going to be quickly checking the status of connection X, but with this implementation the position of X could be different each time I open the tooltip, and depends on the status of Y or Z. I need to scan to find X in one of the lists, then visually backtrack to see what state it is in.

If I have 3 connections, I'd rather the tooltip was always 3 lines high, and each line consistently showed the status of one connection.

axel7083 commented:
> If I have 3 connections, I'd rather the tooltip was always 3 lines high, and each line consistently showed the status of one connection.

I agree, but this is not the design chosen in https://github.com/podman-desktop/podman-desktop/issues/10921 (discussion should be moved their then)

Firewall commented:
Yes, it was not clear from my description but I would also have 1 line for each provider. 

Sorting I would do it the same as on the provider page.

SoniaSandler commented:
@Firewall @deboer-tim @axel7083 I've updated the PR based on the reviews and requests, separated the providers in the tooltip to be one per line, and added a prop to hide the tooltip

jeffmaury reviewed (approved):
LGTM

Firewall reviewed (approved):
Works great now! Thanks

deboer-tim commented:
Thanks for the changes @SoniaSandler !
